flocx_market/common/exception.py

```diff
@@ -0,0 +1,34 @@
+class MarketplaceException(Exception):
+    msg_fmt = "An unknown exception occurred."
+    code = 500
+    safe = False
+
+    def __init__(self, message=None, **kwargs):
+        self.kwargs = kwargs
+        if 'code' not in self.kwargs:
+            try:
+                self.kwargs['code'] = self.code
+            except AttributeError:
+                pass
+
+        if not message:
+            try:
+                message = self.msg_fmt.format(**kwargs)
+            except Exception as e:
```

Comment:
I would prefer to see us catching *specific* exceptions here. It's probably just going to be `KeyError`, right?